gladys-assistant/docker-compose.yml

```diff
@@ -0,0 +1,20 @@
+version: "3.7"
+
+services:
+  web:
+    image: gladysassistant/gladys:v4
+    restart: on-failure
+    stop_grace_period: 1m
+    privileged: true
+    network_mode: host
+    cgroup: host
+    volumes:
+      - ${APP_DATA_DIR}/gladysassistant:/var/lib/gladysassistant
+      - /var/run/docker.sock:/var/run/docker.sock
+      - /dev:/dev
```

Comment:
This is really weird.

Reply:
Binding the entire `/dev` directory is something we probably have to do here so that Gladys Assistant can communicate with any USB that may need to be plugged into the user's device (e.g., a USB dongle for zigbee support).

We can't know in advance what devices a user does or does not have plugged in, so we need to mount the entire `/dev` directory. We're definitely open to suggestions on how to get around this though, as the goal should always be to restrict a container's access and permissions as much as possible.

This device issue is pretty tricky though. For example, let's just assume that the only device a user ever needs to plug in to work with Gladys gets connected as `/dev/ttyACM0`. The logical solution in that case would seem to be to use this bind mount:
`/dev/ttyACM0:/dev/ttyACM0`

But there is a huge problem with this in Docker, which is that if the user does not have a device plugged in at `/dev/ttyACM0` then Docker will error out and exit when trying to bring up the container because the `/dev/ttyACM0` file doesn't exist. This means that users who install Gladys without a usb installed at that file will get a broken install (Gladys won't even start), and any user that starts up Gladys after removing that USB device will also have a broken installation.

Reply:
Exactly. We need to connect to existing or new plugged usb devices and there is no other way to do it.

Home Assistant is doing the same here https://github.com/getumbrel/umbrel-apps/blob/master/home-assistant/docker-compose.yml#L11